src/coreclr/jit/unwindamd64.cpp

```diff
@@ -423,7 +423,15 @@ void Compiler::unwindSaveRegWindows(regNumber reg, unsigned offset)
             code             = (UNWIND_CODE*)&func->unwindCodes[func->unwindCodeSlot -= sizeof(UNWIND_CODE)];
             code->UnwindOp   = (genIsValidFloatReg(reg)) ? UWOP_SAVE_XMM128_FAR : UWOP_SAVE_NONVOL_FAR;
         }
-        code->OpInfo          = (BYTE)(genIsValidFloatReg(reg) ? reg - XMMBASE : reg);
+        if (genIsValidFloatReg(reg))
+        {
+            code->OpInfo = reg - XMMBASE;
```

Comment:
I think @BruceForstall was also suggesting we validate that it's within the expected range to cover future expansions we might have

So we currently have up to 32 for both float and integer registers.

Reply:
Yeah, I was thinking something like:
```
        unsigned unwindRegNum;
        if (genIsValidFloatReg(reg))
        {
            unwindRegNum = reg - XMMBASE;
        }
        else
        {
            assert(genIsValidIntReg(reg));
            unwindRegNum = reg;
        }
        assert(unwindRegNum <= 15);
        code->OpInfo = (UCHAR)unwindRegNum;
        assert((unsigned)code->OpInfo == unwindRegNum); // check that it fit
```

Reply:
done